### DISCUSSION TITLE
django_ory_auth calling /sessions/whoami for every request

### DISCUSSION THREAD

**Discussion by leodotcloud** [ory/awesome-ory#83] [Q&A]
Hi everyone,

Trying out the example [django-ory-cloud](https://github.com/ory/examples/tree/master/django-ory-cloud) and also looking at the source [code](https://github.com/gen1us2k/django_ory_auth/blob/master/django_ory_auth/backend.py#L15) I see that there is an API request made for every app request. This is quite an overhead to use Ory. Am I missing something?

Quick walkthrough of the source code of the Go SDK as well as the other sample projects using Javascript; it's the same behavior.

References:
- https://www.ory.sh/secure-django-app-using-ory/
- https://hackernoon.com/how-to-secure-your-flask-application-with-ory-kratos-and-ory-keto

**Comment by vinckr** [comment]
Hello @leodotcloud 

Please have a look at the [security model](https://www.ory.sh/docs/security-model) that Ory uses by default.
You can now also [convert Ory Sessions to JSON Web Tokens](https://www.ory.sh/docs/identities/session-to-jwt-cors) if you want to reduce the number of API calls.

**Reply by boosh** [reply]
Your [django docs](https://www.ory.sh/secure-django-app-using-ory/) don't work. And I also abandoned Ory due to the performance hit mentioned above. I suggest you rewrite using JWTs if you want anyone to use it.
